src/DI/RedisExtension.php

```diff
@@ -95,11 +97,20 @@ public function beforeCompileStorage(): void
 				throw new RuntimeException(sprintf('Please install nette/caching package. %s is required', IStorage::class));
 			}
 
-			$builder->getDefinitionByType(IStorage::class)
-				->setAutowired(false);
+			// Set autowired off for default nette Caching storage
+			if($storages === 0) {
+				$builder->getDefinitionByType( IStorage::class )
+				        ->setAutowired( false );
+			}
+
+			// Set autowired only default redis connection
+			$autowired = $name === "default";
 
 			$builder->addDefinition($this->prefix('connection.' . $name . 'storage'))
-				->setFactory(RedisStorage::class);
+				->setFactory(RedisStorage::class, ["client" => $builder->getDefinition($this->prefix('connection.' . $name . '.client'))]) // Inject Redis client by connection
```

Comment:
> I don't understand when it fails. Can you clarify it more? And also include some little tests? Thank you.
Here is injected only default client witch is autowired, no client for specific instance of client.

In my app i want have 3 redis storage (databases) because 1 is for app cache, 2 is for sessions and 3 is for aws files. When i deploy app, i flush only app redis database, not aws.